src/app-e-heat-input-from-gas-workspace/app-e-heat-input-from-gas-workspace.service.ts

```diff
@@ -81,26 +84,45 @@ export class AppEHeatInputFromGasWorkspaceService {
       );
     }
 
-    let entity = this.repository.create({
-      id: historicalRecordId ? historicalRecordId : uuid(),
-      appECorrTestRunId,
-      monitoringSystemId: system.id,
-      gasVolume: payload.gasVolume,
-      gasGCV: payload.gasGCV,
-      gasHeatInput: payload.gasHeatInput,
-      userId,
-      addDate: timestamp,
-      updateDate: timestamp,
-    });
+    let entity;
 
-    await this.repository.save(entity);
-
-    entity = await this.repository.getAppEHeatInputFromGasById(entity.id);
+    if (trx) {
+      entity = trx.getRepository(AppEHeatInputFromGas).create({
+        id: historicalRecordId ? historicalRecordId : uuid(),
+        appECorrTestRunId,
+        monitoringSystemId: system.id,
+        gasVolume: payload.gasVolume,
+        gasGCV: payload.gasGCV,
+        gasHeatInput: payload.gasHeatInput,
+        userId,
+        addDate: timestamp,
+        updateDate: timestamp,
+      });
+
+      await trx.getRepository(AppEHeatInputFromGas).save(entity);
+      entity = await trx.getRepository(AppEHeatInputFromGas).findOneBy({id: entity.id});
+    } else {
+      entity = this.repository.create({
+        id: historicalRecordId ? historicalRecordId : uuid(),
+        appECorrTestRunId,
+        monitoringSystemId: system.id,
+        gasVolume: payload.gasVolume,
+        gasGCV: payload.gasGCV,
+        gasHeatInput: payload.gasHeatInput,
+        userId,
+        addDate: timestamp,
+        updateDate: timestamp,
+      });
+
+      await this.repository.save(entity);
+      entity = await this.repository.getAppEHeatInputFromGasById(entity.id);
+    }
```

Comment:
This should use the `withTransaction` method so that the `getAppEHeatInputFromGasById` method can be used when in a transaction:
```suggestion
    const repository = withTransaction(this.repository, trx);
    entity = repository.create({
      id: historicalRecordId ? historicalRecordId : uuid(),
      appECorrTestRunId,
      monitoringSystemId: system.id,
      gasVolume: payload.gasVolume,
      gasGCV: payload.gasGCV,
      gasHeatInput: payload.gasHeatInput,
      userId,
      addDate: timestamp,
      updateDate: timestamp,
    });

    await repository.save(entity);
    entity = await repository.getAppEHeatInputFromGasById(entity.id);
```